taskiq/depends/progress_tracker.py

```diff
@@ -0,0 +1,99 @@
+import enum
+from typing import Generic, Union, overload
+
+from pydantic.generics import GenericModel
+from taskiq_dependencies import Depends
+from typing_extensions import TypeVar
+
+from taskiq.context import Context
+
+# TODO: PEP 696
+# _ProgressType = TypeVar("_ProgressType", default=Any) #  noqa: E800
+_ProgressType = TypeVar("_ProgressType")
+
+
+class Undefined(enum.Enum):
+    """For undefined value."""
+
+    undefined = ...
+
+
+undefined = Undefined.undefined
+
+
+class TaskState(str, enum.Enum):  # noqa: WPS600
+    """State of task execution."""
+
+    PENDING = "PENDING"
+    RECEIVED = "RECEIVED"
+    STARTED = "STARTED"
+    FAILURE = "FAILURE"
+    SUCCESS = "SUCCESS"
+    RETRY = "RETRY"
+
+
+class TaskProgress(GenericModel, Generic[_ProgressType]):
+    """Progress of task execution."""
+
+    state: Union[TaskState, str]
+    meta: _ProgressType
+
+
+class ProgressTracker(Generic[_ProgressType]):
+    """Task's dependency to set progress."""
+
+    def __init__(
+        self,
+        context: Context = Depends(),  # noqa: WPS404
+    ):
+        self.context = context
+
+    @overload
+    async def set_progress(
+        self,
+        state: TaskProgress[_ProgressType],
+    ) -> None:  # pragma: no cover
+        """
+        Set progress of task execution.
+
+        :param state: TaskProgress value.
+        """
+        ...
+
+    @overload
+    async def set_progress(
+        self,
+        state: Union[TaskState, str],
+        meta: _ProgressType,
+    ) -> None:  # pragma: no cover
+        """
+        Set progress of task execution.
+
+        :param state: TaskState value.
+        :param meta: progress of task.
+        """
+        ...
+
+    async def set_progress(
+        self,
+        state: Union[TaskProgress[_ProgressType], TaskState, str],
```

Comment:
I guess, leaving only TaskState or str is sufficent for any usecase.

Reply:
Personally I would not allow str values, because it adds ambiguity. Since, if you check results you want to know what to check. Adding string states adds complexity for every check. 

But on the other hand, I see why people may want it.

Reply:
@taskiq-python/core-team, what're your thoughts on this?

Reply:
```
@broker.task
await def task(progress: ProgressTracker[int] = Depends()): 
    await progress.set_progress(123)
```

Reply:
Okay. I guess I overcomplicate things.  We can just check if it's equals to some value.

```python
task = await my_task.kiq()
# Here we cannot say what type parameter does progress have.
progress = task.get_progress()
if progress.state == "progress":
	print("Current progress", 123)

```